rust-libretro/src/environment.rs

```diff
@@ -474,14 +520,17 @@ pub unsafe fn get_rumble_interface(
 /// 0 in [`retro_input_state_t`].
 /// Example bitmask: `RetroDevice::JOYPAD | RetroDevice::ANALOG`.
 #[proc::context(RunContext)]
-pub unsafe fn get_input_device_capabilities(callback: retro_environment_t) -> RetroDevice {
-    // I’m not entirely sure why this call returns a 64 bit value when the `RETRO_DEVICE_MASK` allows only eight distinct types.
+pub unsafe fn get_input_device_capabilities(
+    callback: retro_environment_t,
+) -> Result<RetroDevice, EnvironmentCallError> {
     // uint64_t *
-    if let Some((caps, _)) = get::<u64>(callback, RETRO_ENVIRONMENT_GET_INPUT_DEVICE_CAPABILITIES) {
-        return RetroDevice::from_bits_truncate(caps as u8);
-    }
+    let caps = get::<u64>(callback, RETRO_ENVIRONMENT_GET_INPUT_DEVICE_CAPABILITIES)?;
+
+    // I’m not entirely sure why this call returns a 64 bit value when the `RETRO_DEVICE_MASK`
```

Comment:
I think the return value is 64-bit to allow extra types of input devices to be added without breaking ABI compatibility.